x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -913,7 +913,6 @@ Collection<Object> createComponents(
         DestructiveOperations destructiveOperations = new DestructiveOperations(settings, clusterService.getClusterSettings());
         final RemoteAccessAuthenticationService remoteAccessAuthcService = new RemoteAccessAuthenticationService(
             clusterService,
-            apiKeyService,
```

Comment:
On second thought, the dependency here was unnecessary: it was only used to ensure the service was enabled but that happens implicitly during authentication, and avoiding this dependency and making the ApiKeyService parsing functionality static makes things cleaner all in all.